ui/src/actions/login.js

```diff
@@ -77,47 +66,21 @@ export function startSession(userInControl) {
   };
 }
 
-export function refreshSession() {
-  return (dispatch) =>
-    fetch('mxcube/api/v0.1/login/refresh_session', {
-      method: 'GET',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
-      },
-      credentials: 'include',
-    });
-}
-
 export function getLoginInfo() {
   return (dispatch) =>
-    fetch('mxcube/api/v0.1/login/login_info', {
-      method: 'GET',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
+    fetchLoginInfo().then(
+      (loginInfo) => {
+        dispatch(setLoginInfo(loginInfo));
+        return loginInfo;
       },
-      credentials: 'include',
-    })
-      .then((response) => response.json())
-      .then(
-        (loginInfo) => {
-          dispatch(setLoginInfo(loginInfo));
-          return loginInfo;
-        },
-        () => {
-          dispatch(showErrorPanel(true));
-          dispatch(setLoading(false));
-        },
-      );
-}
-
-export function signOut() {
-  localStorage.setItem('currentUser', '');
-  return { type: 'SIGNOUT' };
+      () => {
+        dispatch(showErrorPanel(true));
+        dispatch(setLoading(false));
+      },
+    );
 }
 
-export function signIn(proposal, password, navigate) {
```

Comment:
Trying to keep the naming consistent with the API. The `do` prefix helps distinguish between the Redux action creator and the new fetching function `logIn` in `/src/api/login.js`.

Reply:
Its not at all important and we can go with the `do` prefix, but I have to admit that I'm not a big fan of it :) (That is just my opinion :) ). Is it a common prefix to use in the React world ?

Reply:
Not at all, it's just a general way to avoid naming conflicts with functions. More of a temporary work around in this case, as my goal is to see if I can remove this `doSignIn` action creator eventually.

Reply:
I should add that the main reason for having unique function names even across files is to facilitate search-and-replace operations.

That's why I also made sure to be consistent inside `mapDispatchToProps`: `doLogIn: bindActionCreators(doLogIn, dispatch)`. Otherwise, it's hard to find where actions are dispatched.

Reply:
Back to `doLogIn`, the problem is also that I couldn't find a better name for the `logIn` API function.

For `GET` requests that fetch resources, I'm planning on using the `fetch` prefix - e.g. `fetchLoginInfo`, but for API functions that send mutation requests, it's more complicated:

- using the HTTP method as prefix is a probably a bad idea: `getLoginInfo` (conflicts with the Redux action creator), `getRefreshSession` (makes no sense), `getSignOut` (same)...
- CRUD prefixes (`create`, `read`, `update`, `delete`) are way too generic;
- the `send` prefix is perhaps the most promising: `sendLogin`, `sendSignOut`, `sendRefreshSession` ... perhaps combined with CRUD prefixes for REST-compliant endpoints: `sendCreateSession` (fake example) — of course we could make an exception for `sendFeedback` to avoid `sendSendFeedback`...

Reply:
Sounds very reasonable :), I also like `send` and `fetch` for the reason that you mention above.